### PR TITLE
[CLI] replace host/port CLI args with full URL

### PR DESCRIPTION
**Pull request by christinacelee**
## Motivation

In CLI, fully replace host/port args with passing in full URL, to give user full customizability of URL options (e.g. http protocol type, using paths vs ports)

## Test Plan

Tested on local swarm CLI and devnet

### CODE REVIEW

File: scripts/cli/start_cli_testnet.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ source "$HOME/.cargo/env"
 
 SCRIPT_PATH="$(dirname $0)"
 
-RUN_PARAMS="--host https://client.testnet.libra.org --port 443 "
+RUN_PARAMS="--url https://client.testnet.libra.org:443 "
```
Comment (thefallentree) [inline scripts/cli/start_cli_testnet.sh]: No need for :443

File: docker/client/client.Dockerfile [hunk 2]
```diff
@@ -26,7 +26,7 @@ RUN mkdir -p /opt/libra/bin /opt/libra/etc
 COPY --from=builder /libra/target/release/cli /opt/libra/bin/libra_client
 
 ENTRYPOINT ["/opt/libra/bin/libra_client"]
-CMD ["--host", "ac.testnet.libra.org", "--port", "8000"]
+CMD ["--url", "http://client.testnet.libra.org:80"]
```
Comment (thefallentree) [inline docker/client/client.Dockerfile]: No need for :80

File: client/cli/src/client_proxy.rs [hunk 3]
```diff
@@ -148,7 +150,9 @@ impl ClientProxy {
 
         let faucet_server = match faucet_server {
             Some(server) => server,
-            None => host
+            None => url
+                .host_str()
+                .ok_or_else(|| format_err!("Missing host in URL"))?
                 .replace("http://", "")
                 .replace("https://", "")
```
Comment (metajack) [inline client/cli/src/client_proxy.rs]: These aren't needed now.

File: client/cli/src/client_proxy.rs [hunk 4]
```diff
@@ -148,7 +150,9 @@ impl ClientProxy {
...
                 .replace("client", "faucet"),
```
Comment (metajack) [inline client/cli/src/client_proxy.rs]: Does this normally appear in the host position?

File: docker/README.md [hunk 5]
```diff
@@ -10,4 +10,4 @@ This directory contains [Docker](https://www.docker.com/) configuration and scri
 4. To test the validator image locally, run `docker/validator/run.sh`
 5. To test the faucet server image locally, run `docker/mint/run.sh`
 6. Run the client as follows:
-   `cargo run -p cli --bin cli -- -a localhost -p 8000 -f localhost:8080`
+   `cargo run -p cli --bin cli -- -u http://localhost:8000 -f localhost:8080`
```
Comment (metajack) [inline docker/README.md]: Should `-f` also use urls?

Comment (christinacelee) [reply docker/README.md]: Can do this in another PR

File: client/cli/src/client_proxy.rs (hunk 4 above)
Comment (christinacelee) [reply client/cli/src/client_proxy.rs]: yes - for this branch we expect a very specific URL

**Pull request closed without merging**